src/library_int53.js

```diff
@@ -123,6 +123,19 @@ mergeInto(LibraryManager.library, {
     return (lo >>> 0) + (hi >>> 0) * 4294967296;
   },
 
+  // Given a signed i53 number, emit the low 32 bits. This is useful for
+  // calling into wasm when legalization is needed, and then instead of just
+  // doing f(num) you would do f(sendI53BitsLow(num), sendI53BitsHigh(num)).
+  $sendI53BitsLow: function(num) {
+    // TODO: assert on num being in the valid i53 range
+    return num | 0;
+  },
+  $sendI53BitsHigh: function(num) {
+    // TODO: assert on num being in the valid i53 range
+    num = Math.floor(num / 4294967296);
+    return num | 0;
```

Comment:
Do we use the `send` and `receive` naming still widely? I always thought those terms confusing, since from static reading of an individual function, it is not clear where the data is supposed to be flowing, and what side should be considered the sender, and which one should be regarded the receiver.

I know there is a comment there to explain, but it seems like the name choice makes the comment mandatory?

It would be nice to avoid send and receive terms if possible. Maybe these could be named `getLowI32Part(num)` and `getHighI32Part(num)` instead?

The impl could read `Math.floor(num / 4294967296) | 0;` without loss on readability?

For the TODO, maybe expanding the comment with

```js
// You can use assert(Number.isSafeInteger(num)); to verify that a given number
// can be converted without precision loss.
```
could be used, so the caller can emit or not emit such an assert, depending on their use case whether they accept lossy conversion or not?

Reply:
Thanks for the feedback! Meanwhile though I just removed all this code for now. But we may get back to it as an optimization later.

I do agree the names are ambiguous. Perhaps `forCallToWasm` or such would make it more clear..?

Reply:
Whats wrong with just `splitI53`?